docs/examples/regression.py

```diff
@@ -179,7 +179,7 @@
 # these parameters by optimising the marginal log-likelihood (MLL).
 
 # %%
-negative_mll = jit(gpx.objectives.ConjugateMLL(negative=True))
+negative_mll = gpx.objectives.ConjugateMLL(negative=True)
 negative_mll(posterior, train_data=D)
```

Comment:
So, I am guessing the cite function as it stands does not work on traced objects? 

- (1) Is it possible to get it to work here in future? And if so open an issue for this.
- (2) If not, what does `cite` currently return? Can we check if the function is traced, and return a helpful error e.g., `TypeError('cite does not work on jit traced objects, please use non-jitted one')`?

We could build either (1) or (2) in to the `dispatch` decorator.

Reply:
I can open an issue for (1) - it may be possible, but I don't know how :/ I'll do your suggestion for 2 in the revision.